go/mcap/range_index_heap.go

```diff
@@ -0,0 +1,41 @@
+package mcap
+
+type rangeIndex struct {
+	chunkIndex        *ChunkIndex
+	messageIndexEntry *MessageIndexEntry
+	reverse           bool
+	buf               []uint8
+}
+
+func (ri *rangeIndex) logTime() uint64 {
+	if ri.chunkIndex != nil {
+		if ri.reverse {
+			return ri.chunkIndex.MessageEndTime
+		}
+		return ri.chunkIndex.MessageStartTime
+	}
+	return ri.messageIndexEntry.Timestamp
+}
+
+type rangeIndexHeap []rangeIndex
+
+func (h rangeIndexHeap) Len() int      { return len(h) }
+func (h rangeIndexHeap) Swap(i, j int) { h[i], h[j] = h[j], h[i] }
+func (h *rangeIndexHeap) Push(x interface{}) {
+	*h = append(*h, x.(rangeIndex))
+}
+
+func (h *rangeIndexHeap) Pop() interface{} {
+	old := *h
+	n := len(old)
+	x := old[n-1]
```

Comment:
I'm confused by this because the documentation at https://pkg.go.dev/container/heap says:

>The minimum element in the tree is the root, at index 0.

Reply:
This also confuses me, since this is a verbatim copy from the IntHeap example there.

Reply:
This is confusing because it's the `Pop()` method of `heap.Interface`, which has the comment:  // remove and return element Len() - 1.